### DISCUSSION TITLE
"Error: Exit code: 1" when trying to download videos - help please!

### DISCUSSION THREAD

**Discussion by Anoneemouse** [aclap-dev/video-downloadhelper#1381] [Help and Q&A]
### Your question

Hi,
I'm trying to download videos from this site: https://vimeo.com/showcase/11413698
There are 8 videos on it that I want (one of each horse), in the highest resolution/file size possible (apparently 1920x1080). For some reason, 16 videos show up in the VDH download icon, and I have no way to tell which is which.

The first one I tried downloaded just fine. I waited the mandatory 2 hours, and tried the second one, only to get this error:
~~~~
Exit code: 1
Error: Exit code: 1

    at ba.receive (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:13152)
    at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42681
~~~~~

I tried reloading the page, but got the same error again. Then I tried relaunching the page from the email link I had to it, and was able to download another video. After another 2 hours, I tried for a third, only to get that same "error code 1" message, several times. I tried reloading the page, and relaunching the page, but I keep getting that same error code.

Thanks so much in advance for help with this problem!!!  

I'm using the most current version of Chrome on a current version of Windows 10. I've pasted the VDH info below also. 

version: 9.1.1.0
channel: stable
build date: 2024-10-15
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\noney",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

### Addon & CoApp version and details

version: 9.1.1.0
channel: stable
build date: 2024-10-15
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\noney",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by xezoxeon** [comment]
as of today its still not working

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 24H2 build 26100.2033, Firefox 131.0.3 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.

The significant difference between your setup & mine is probably the browser.  You're on Chrome & I'm on Firefox.  That means I have access to a VDH beta & you do not.  This would not be the first time something worked better on Firefox than it does on Chrome.

I downloaded all 8 videos.  I would have downloaded them simultaneously but they're all so short that generally the one I had just launched finished while I was launching the next one.  They all downloaded without incident & they all played fine in VLC with both video & audio.  I didn't actually sit & watch them, just skimmed them to verify they were complete & would play.

I have to say that just the little bits I watched made me think they were all the same.  Maybe a horse connoisseur could tell them apart but they all looked the same to me.  If it weren't for the different labels at the start of each video, I would have thought I downloaded the same video 8 times.

Usually, you can tell which video matches which entry in the VDH menu from the durations.  But some of these videos have the same durations.  Others have complained about this inability to distinguish one video from another on a page like this.  I have to say I don't know exactly how VDH is supposed to differentiate these videos.  If there is information in the HTML coding of the page that puts a unique label on each video, VDH could use that.  I don't know enough about these things to determine whether that is the case on this web page.  And even if I did, I wouldn't know how to change VDH, being, like I said, just another user.

I tried popping up a context menu on each video.  Normally, there is an entry in the context menu that allows you to cascade out a submenu with VDH commands on it.  But that's not what I got here.

![image](https://github.com/user-attachments/assets/f04fca76-7c85-43d3-922b-e0d567d1ed32)

However, you can see I did get an entry for `Open Frame in New Tab`.  Using that did open a new tab with just that individual video in it.  You could go through the (admittedly somewhat tedious) process of opening each video in its own browser tab & using VDH there to download the video.  You could then know which one you were downloading & give it a name that would be more descriptive than the default that VDH offers . . . which is the same name for every video.

Strangely, when I did isolate a video into its own tab, the popup context menu did include the VDH submenu.

![image](https://github.com/user-attachments/assets/591d1bd4-de66-474e-bda7-619d18dc2959)

I have no idea why this happened.  You'd think the context menu would be the same either way.  Strangely again, the context menu did not contain the usual entry for `Save video as ...` that I've seen so often elsewhere.  I can't explain this either.

I don't know if any of this is applicable to you.  I don't know if Chrome offers the same kind of popup context menus that Firefox does.  Also, the fact that I could download these things no problem with Firefox doesn't really help you on Chrome.  You don't have a license so you wouldn't be losing anything to try things with Firefox.  There's no 2-hour waiting period on Firefox with unlicensed VDH.

**Reply by Anoneemouse** [reply]
Thanks so much for the response, GrampaWildWilly! Unfortunately Chrome doesn't have that "open frame in a new tab" option - I wish to heck it did!  I can view the source, but don't see a link to the file in the source code - but didn't even think to look at the url as Mattmjs - he's got a great work around for that which works!!!  

If you're interested, they are all different horses. They're all young Standardbred racehorses (one year old - e.g., yearlings), who are going in a large auction. So everyone uses the same format - e.g., have the youngsters go speeding along the fence line in a trot so prospective buyers can see their build, stride, and movement... and then some runs straight from the front and rear, so you can see if the flight path of their legs/feet are straight or if they twist at all (makes them more prone to lameness). Of course each video also has the little intro sequence that shows the horse's hip number (e.g., the order they'll go through the auction ring) plus their name and the name of their sire & dam (Pop & Mom).  :0)  The easiest difference to note for someone not used to evaluating horses is different coat colors, sometimes markings (like white on the lower leg or face), etc. But they all look very different to me as I've worked with horses for many years, tho never directly with Standardbred racehorses (harness racing, e.g., with the sulky/bike behind the horse with a driver, rather than a jockey on their backs).

I've got to figure out how to put two different versions of firefox on my darned computer without messing up the older version & it's user profile... I like firefox better, but have an old version of firefox that I keep because I use it's ftp extension to edit a website for a friend... So I have to either figure out how to have two different versions of firefox on the same computer (which I've done once many years ago - but don't remember how to set it up now!)... or I've got to join the present day and get/use/set up a different ftp program and then I could ditch the old firefox version and just have a current one on the system. I'm sure open to suggestions if any of you happen to know off the top of your head how to put a second version on the system without messing up the older one... or have suggestions for a good free ftp program. :0)

I didn't realize that there's no 2hr delay for HLS downloads when using firefox! That in and of itself is a real incentive to figure out a way to put a second/current firefox version on the computer. Seems odd that VDH is different with chrome v. firefox that way. 

> _"Usually, you can tell which video matches which entry in the VDH menu from the durations. But some of these videos have the same durations. Others have complained about this inability to distinguish one video from another on a page like this. I have to say I don't know exactly how VDH is supposed to differentiate these videos. If there is information in the HTML coding of the page that puts a unique label on each video, VDH could use that."_

In the VDH dropdown menu, I didn't realize that number right after the file resolution was the duration, but that does make sense. As you note, however, a number of them have the same duration. I can tell you that each video has to have a different name in the coding - that's the only way the code can tell which video to put where. VDH used to do a MUCH better job of picking up the video titles, but then awhile back when there was a major upgrade, I think when they added the side downloads etc., it got much worse at displaying the video titles and also at putting the detailed video info into the title bar when you're downloading a file.  It's a shame, because it was so much more useful when it picked up video titles and automatically put them in the file name as you were saving a file, and displayed them better in the dropdown menus also.  But VDH is still awesome anyhow. :0)  Here's what I see in the dropdown for this page:

![image](https://github.com/user-attachments/assets/2e2dd6dc-ddb3-49f0-a6d8-e64dec34f027)

Anyhow, thank you so much for your reply and all the ideas in it!

**Reply by Mattmjs** [reply]
There is a portable version of Firefox you can put on a removable drive from the portable apps site.

**Reply by GrampaWildWilly** [reply]
That looks like a really old version of VDH.  On the new version, the video duration is superimposed on the icon in the VDH menu.  That is, when VDH can detect a duration.  Apparently, there are certain ways of coding HTML that don't allow VDH to detect a duration.

For multiple Firefoxes, you need to install from the Mozilla web site.  Download a version of the Firefox installer to your hard drive.

![#01](https://github.com/user-attachments/assets/606a511a-be62-4233-989b-bb5e194ee9db)
![#02](https://github.com/user-attachments/assets/1df07db7-3033-443c-bfdf-032aee2380e3)
![#03](https://github.com/user-attachments/assets/ac6e38b8-6691-45be-ac1e-126986890363)


Then during the installation, a page comes up that offers standard or custom install.  Select custom.  That will give you a page where you can specify a different directory for Firefox to install into.  Otherwise, just let the install go like it was a standard install.  You can install however many Firefoxes you like that way.  Interestingly, if you go into about:profiles, it will list the profiles for all the versions of Firefox.  That's because they all share the same directory.  But the profiles have different names so it does keep them separate.

Seems to me I recall reading in the Firefox release notes a couple of years ago that they finally removed all the FTP support from the browser.  I haven't encountered an FTP site in years.  But if you want an FTP client, you should just do a web search for something.  Just for grins & giggles, I opened a command prompt just now & entered `ftp`.  It worked.  So there's an FTP client built into Windows (I'm running Windows 11 Pro 64-bit version 24H2 build 26100.2033).  In whatever way Mozilla may have deprecated & deimplemented FTP, it's apparently still a standard part of the TCP/IP stack delivered with Windows.  I remember using the command line FTP many years ago.  It takes a little getting used to but once you get the hang of it, it's pretty easy to deal with.  I've completely forgotten everything I used to know about it so I would need to go through the learning curve again.  Except for the minor detail that I don't have any reason to.

**Reply by GrampaWildWilly** [reply]
Horses.  Meh.  I think I may have been irrevocably damaged by what my daddy used to say: "It takes a horse's ass to put $2 down on a horse's nose."  I think I might be a bit more tolerant than he was but I don't think I've totally recovered.  To me, a horse is a horse . . . unless it talks.

**Reply by GrampaWildWilly** [reply]
Here's another way you can go about this.  I've just gotten the 10 videos off this page:

https://www.metopera.org/season/2024-25-season/rigoletto/

The steps were these:

1. Load the page in Firefox.
2. In the VDH `More settings`, scroll all the way to the bottom & click `Reload addon`.
3. Launch the first video & let it play a second.  This will get VDH to recognize that video & it will be the only entry on the VDH menu..
4. Do a `Download As` on that video.  Name it whatever you'd like.
5. Once the download completes, do another `Reload addon`.  Alternatively, you could just delete the entry from the VDH menu.  The point is to make the VDH menu be empty.  `Reload addon` seems like the lazy boy's way of doing things so that's my choice.
6. Launch the next video & let it play a second.  Now it will be the only entry on the VDH menu.
7. Rinse & repeat for each video on the page.

**Reply by GrampaWildWilly** [reply]
Hmmmmm..........  For some reason, that approach doesn't work on your horsey page.  That's annoying.  I have no idea why not.

**Reply by GrampaWildWilly** [reply]
On the other hand . . .  Below each horsey video there is a number.  Click that.  That takes you to a page that consists of just that video.  VDH can then download just that one video & you can name it as you please.  Obviously, this is not a general technique one could use on any web site.  This particular web page offers this architecture.  Seems like the answer to your problem.

**Comment by Mattmjs** [comment]
Firefox looks to be unique on open frame in new tab / window. I've looked in Edge and Ungoogled chromium and no option.
The only thing closest to that is open view frame source. This opens a page with the code for the video.

In the url bar it'll have `view-source:https://` with vimeo link. All you have to do backspace is `view-source:` and press enter.

**Reply by Anoneemouse** [reply]
Mattmjs, THANK YOU THANK YOU THANK YOU!! Before I'd even posted here, I'd tried looking at the frame source code and couldn't see the video file source there... but DUH, it never even occurred to me to look at the page url as you're suggesting - that works beautifully to get each video onto it's own page where at least then I'm sure which I'm downloading! I'd already been really frustrated when I tried just picking a different file position - and still managed to get the same horse's video three different times! (with a two hour wait between each try). 

In a few days or so they'll post these files differently so selecting one will pop it onto it's own page - but I really wanted to get these videos now... And now with the combination of GrampaWildWilly and yours, I can at least get each on it's own page so I don't accidently wind up downloading the same video multiple times while not getting some others at all. 

Thanks again!

**Comment by xezoxeon** [comment]
Download failed


Sadly we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error (it's anonymous!).

**Reply by Anoneemouse** [reply]
Hi Xezoxeon - report it to who, where?  I'll be happy to report it if you just tell me where to do so. The funny thing is that the download error seems to be intermittent. I've been able to download a video at times, but then I start getting the error... then later I manage to get another video downloaded. And I can't tell if anything I'm doing has anything to do with when it does or doesn't work. By the way, in this case there are no worries about it being anonymous - the owners of those videos don't mind people downloading them. Most years they have a download button for each video even, so I'm a bit surprised that they don't have one this year, although there was one other year that they didn't and then later they added the download option.

**Reply by Mattmjs** [reply]
That's just the generic failed download message. I've already tested the soaper site which @xezoxeon uses.
As of the moment still no fix, but if you use VLC it will handle it while you wait for the fix.
